docs/REST.md

```diff
@@ -364,66 +367,61 @@ Calling `getUsername("ID")` will make a HTTP request on path `users/ID/name`.
-It takes an obligatory `name` argument that specifies HTTP header name.
+It takes an obligatory `name` argument that specifies HTTP header name (case insensitive).
 
-Values of header parameters are serialized into `HeaderValue` objects,
-see [serialization](#path-query-and-header-serialization) for more details.
+Values of header parameters are serialized into `HeaderValue` objects. This means that the macro engine will look for
+implicit instances of `AsRaw[HeaderValue, T]` and `AsReal[HeaderValue, T]` for every header parameter of type `T`.
+See [serialization](#path-query-and-header-serialization) for more details.
 
 ### Body parameters
 
-As mentioned earlier, every parameter of non-`GET` API trait method is interpreted either as a field
-of a JSON object sent as HTTP body or (when method is annotated as [`@FormBody`](#formbody)) as a part of URL-encoded
-form sent as HTTP body. Just like for path, query and header parameters, there is a
-`@BodyField` annotation which requests this explicitly. However, it only makes sense to use this
-annotation when one wants to customize the field name because `GET` methods do not accept body parameters.
-A method annotated as `@GET` having a parameter annotated as `@BodyField` will be rejected by REST
-macro engine.
+Every parameter of an API trait method (except for `@GET`) is interpreted as a field of a JSON object sent as 
+HTTP body. Just like for path, query and header parameters, there is a `@Body` annotation which requests this 
+explicitly. However, the only reason to use it explicitly is in order to customize the name of JSON field.
 
-Body parameters are serialized into `JsonValue` objects (by default) or `QueryValue` objects
-(for [`@FormBody`](#formbody) methods), see
-[serialization](#body-parameter-serialization) for more details.
+Body parameters are serialized into `JsonValue` objects. This means that the macro engine will look for
+implicit instances of `AsRaw[JsonValue, T]` and `AsReal[JsonValue, T]` for every path parameter of type `T`.
+See [serialization](#path-query-and-header-serialization) for more details.
 
-### Single body parameters
+#### `@FormBody`
 
-Every non-`GET` API method may also take a single parameter annotated as `@Body` in place of multiple
-unannotated parameters (implicitly interpreted as `@BodyField`s). This way the value of that parameter
-will be serialized straight into HTTP body. This gives you full control over the contents sent in HTTP body
-and their format (i.e. it no longer needs to be `application/json`).
+Non-`GET` methods may be annotated as `@FormBody`. This changes serialization of body parameters
+from JSON object to HTTP form, encoded as `application/x-www-form-urlencoded`. Each body parameter itself
+is then serialized into `QueryValue` rather than `JsonValue`.
+
+#### `@CustomBody`
+
+Methods annotated as `@CustomBody` are required to take **exactly one** body parameter. This body parameter will
```

Comment:
not 100% sure, but `annotated with` is probably more common

Reply:
`s/annotated as/annotated with/g`